src/main/resources/assets/computercraft/lua/rom/modules/main/http/pastebin.lua

```diff
@@ -0,0 +1,128 @@
+--- Parse the pastebin code from the given code or URL
+local function parseCode(paste)
+    local patterns = {
+        "^([%a%d]+)$",
+        "^https?://pastebin.com/([%a%d]+)$",
+        "^pastebin.com/([%a%d]+)$",
+        "^https?://pastebin.com/raw/([%a%d]+)$",
+        "^pastebin.com/raw/([%a%d]+)$",
+    }
+
+    for i = 1, #patterns do
+        local code = paste:match(patterns[i])
+        if code then
+        return code
+        end
+    end
+
+    return nil
+end
+
+-- Download the contents of a paste
+local function download(code)
+    if type(code) ~= "string" then
+        error("bad argument #1 (expected string, got " .. type(code) .. ")", 2)
+    end
+
+    if not http then
+        return false, "Pastebin requires http API"
+    end
+
+    -- Add a cache buster so that spam protection is re-checked
+    local cacheBuster = ("%x"):format(math.random(0, 2 ^ 30))
+    local response, err = http.get(
+        "https://pastebin.com/raw/" .. textutils.urlEncode(code) .. "?cb=" .. cacheBuster
+    )
+
+    if not response then
+        return response, err
+    end
+
+    -- If spam protection is activated, we get redirected to /paste with Content-Type: text/html
+    local headers = response.getResponseHeaders()
+    if not headers["Content-Type"] or not headers["Content-Type"]:find("^text/plain") then
+        return false, "Pastebin blocked due to spam protection"
```

Comment:
Pinging @Lemmmy here to get their thoughts on this. I don't really want to return a massively long error message, but at the same time we need to return informative info to the user within the `pastebin` program.

Reply:
Return a short error message, like `Got HTML response`, then in the `pastebin` program, change that to the longer, more friendly message `Pastebin blocked due to spam protection`.